static/app/utils/queryClient.tsx

```diff
@@ -242,22 +253,15 @@ export function useInfiniteApiQuery<TResponseData>({
   enabled,
   staleTime,
 }: {
-  queryKey: ApiQueryKey;
+  queryKey: InfiniteApiQueryKey;
   enabled?: boolean;
   staleTime?: number;
 }) {
   return useInfiniteQuery({
-    // We append an additional string to the queryKey here to prevent a hard
-    // crash due to a cache conflict between normal queries and "infinite"
-    // queries. Read more
-    // here: https://tkdodo.eu/blog/effective-react-query-keys#caching-data
-    queryKey:
-      queryKey.length === 1
-        ? ([...queryKey, {}, 'infinite'] as const)
-        : ([...queryKey, 'infinite'] as const),
```

Comment:
Now you're required, per TS, to pass in a key that includes the string `'infinite'` at the start. `ApiQueryKey` and `InfiniteApiQueryKey` are not subclasses of each other, but it's possible to convert from one to the other.